### DISCUSSION TITLE
Bouncing ball with spin

### DISCUSSION THREAD

**Discussion by KieDani** [google-deepmind/mujoco#2131] [Asking for Help] [answered]
### Intro

Hi!

I am a researcher at University of Augsburg, I use MuJoCo for my research on sport's analysis using computer vision.

### My setup

MuJoCO 3.2.3, Python API

### My question

I want to simulate the trajectory of a table tennis ball and I am thinking of using MuJoCo for these simulations. I am especially interested in the behaviour of the ball if it is in contact with the plate. Since the rotation of the ball effects its bouncing behaviour significantly, is it possible to simulate this with MuJoCo. Does the rotation have an effect on the bouncing behaviour in MuJoCo? If so, how to finetune this effect?

Thank you for your help! 

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment, marked as answer]
Oh ya, this is definitely possible. [Here is an example model](https://github.com/google-deepmind/mujoco/blob/main/test/engine/testdata/spin_recoil.xml), load the keyframe to see the effect. This model has a ballast inside the ball for the spin recoil effect, I don't think you'll need that for a 🎾. 

The authors of [this paper](https://arxiv.org/abs/2408.03906v1) did quite a bit of MuJoCo system identification on table tennis balls. Contact the second author for details.
